src/gsplat-labels.ts

```diff
@@ -0,0 +1,111 @@
+import { Vec3 } from "playcanvas";
+
+class Category{
+    name: string;
+    id = 0;
+    color: Array<number>;
+    attributes: any[] = [];
+
+    constructor(name: string, id: number, color: Array<number>, attributes: any[]) {
+        this.name = name;
+        this.id = id;
+        this.color = color;
+        this.attributes = attributes;
+    }
+}
+
+class Annotation{
+    id = 0;
+    category_id = 0;
+    attributes: any[] = [];
+    isHidden = false;
+
+    constructor(id: number, category_id: number, attributes: any[], isHidden: boolean) {
+        this.id = id;
+        this.category_id = category_id;
+        this.attributes = attributes;
+        this.isHidden = isHidden;
+    }
+}
+
+class Label{
+    name: string;
+    annotations: Array<Annotation>;
+    point_annotations: Array<number>;
+    category_annotations: Uint32Array;
+
+    constructor(name: string, annotations: Array<Annotation>, point_annotations: Array<number>, category_annotations: Uint32Array) {
+        this.name = name;
+        this.annotations = annotations;
+        this.point_annotations = point_annotations;
+        this.category_annotations = category_annotations;
+
+        // Generate annotation for unlabelled points if one does not exist yet
+        const hasUnlabeledAnnotation = this.annotations.some(ann => ann.category_id === 0);
+        if (!hasUnlabeledAnnotation) {
+            // TODO - do we need to add support for data formats when there may be an existing annotation with id 0?
+            const unlabeled_annotation = new Annotation(0, 0, [], false);
+            // insert at beginning
+            this.annotations.unshift(unlabeled_annotation);
+        }
+    }
+}
+
+class GSplatLabels {
+    static MAX_UNIFORM_COLORS = 4000;
```

Comment:
Shader code does not compile with larger color buffer - should be reworked to use dynamically allocated buffers perhaps?